src/cargo/util/toml.rs

```diff
@@ -540,7 +546,11 @@ impl TomlManifest {
         }
 
         // processing the custom build script
-        let new_build = project.build.as_ref().map(PathBuf::from);
+        let manifest_file = util::important_paths::find_root_manifest_for_wd(None, &layout.root)
```

Comment:
Looking this over again, I'm wondering if I actually have to do this, or if I can just assume `base_dir == layout.root`?